src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -420,16 +431,19 @@ public static IEnumerable<EventSource> GetSources()
         /// <param name="commandArguments">A set of (name-argument, value-argument) pairs associated with the command</param>
         public static void SendCommand(EventSource eventSource, EventCommand command, IDictionary<string, string?>? commandArguments)
         {
```

Comment:
Similarly, rather than all this indenting, could we make it be:
```C#
if (!IsSupported)
{
    return;
}
```
and leave the rest of the method untouched?

Reply:
Also, just to confirm my understanding of the benefit this provides, this scheme let's the linker see that the body of the method is unused, and so even if there's a caller to it, it's able to trim away the body and the references to anything the body in turn uses?

Reply:
> this scheme let's the linker see that the body of the method is unused, and so even if there's a caller to it, it's able to trim away the body and the references to anything the body in turn uses?

Correct. This method is public, so even if someone is calling the method nothing will happen because the body will be empty.

> Does that make a meaningful difference, the method is called but we remove its implementation, and that saves a measurable amount of space?

Its the full chain of methods/types that start getting preserved if we don't. From what I've seen, one method being left in can keep a good chunk of the whole class (and internal classes). For example, this method calls `DoCommand` => `EnsureDescriptorsInitialized` => `CreateManifestAndDescriptors`, which all brings in a ton of stuff - `ResourceManager`, `ManifestBuilder`, etc.